Epam/Java Basics_UpSkill Lab #1/Algorithmization/Decomposition_using_methods/src/com/androsor/decomp/HomeTask14.java

```diff
@@ -0,0 +1,58 @@
+package com.androsor.decomp;
+
+/**
+ * A natural number with n digits is called an Armstrong number if the sum of its digits raised to the power n is equal
+ * to the number itself. Find all Armstrong numbers from 1 to k. To solve the problem, use decomposition.
+ */
+
+public class HomeTask14 {
+
+    public static void main(String[] args) {
+
+        int k = HomeTask1.dataInput(" Введите число К = ");
+        while (k < 0) {
+            System.out.println(" Число должно быть положительным!");
+            k = HomeTask1.dataInput(" Введите число К = ");
+        }
+
+        System.out.printf(" В промежутке от 1 до %d существуют следующие числа Армстронга: \n", k);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.